core/src/main/java/com/graphhopper/util/shapes/Polygon.java

```diff
@@ -192,4 +224,26 @@ public double getMaxLat() {
     public double getMaxLon() {
         return maxLon;
     }
+
+    @Override
+    public String toString() {
+        return "polygon (" + lat.length + " points)";
+    }
+
+    public static Polygon parsePoints(String pointsStr, double growFactor) {
+        String[] arr = pointsStr.split(",");
+
+        double[] lats = new double[arr.length /2];
+        double[] lons = new double[arr.length /2];
+
+        for (int j = 0; j < arr.length; j++) {
+            if (j % 2 == 0) {
+                lats[j / 2] = Double.parseDouble(arr[j]);
+            } else {
+                lons[(j - 1) / 2] = Double.parseDouble(arr[j]);
```

Comment:
here `j/2+1` makes more sense as index into the lons array IMO

Reply:
@karussell Can you give an example of a ``j`` where it is clear how it would make more sense?

Reply:
Oh, probably I misunderstood this index, sorry for the bubble